sumpy/tools.py

```diff
@@ -430,6 +431,26 @@ def diff(self, mi, save_intermediate=lambda x: x):
 
         return result * save_intermediate(1 / self.taker.rscale ** max_order)
 
+
+def diff_transformation(derivative_transformation, variable_idx, variables):
```

Comment:
```suggestion
def diff_derivative_coeff_dict(derivative_coeff_dict, variable_idx, variables):
```
(I don't know really... these dictionaries are starting to proliferate, indicating that they're quite useful. I'm not sure I'm loving "derivative transformation as a name for them. Maybe we should have a separate type for them, even if it's just a one-attribute wrapper around the dictionary?)